dfa/src/main/scala/dfa/DFA.scala

```diff
@@ -1,3 +1,19 @@
 package dfa // leave this line in the file
 
-// TODO: replace this comment with your implementation
+case class State(label: String)
+
+case class Transition(from: State, to: State, symbol: Char)
+
+class DFA(val states: Set[State], val transitions: Set[Transition], val start: State, val accept: Set[State]) {
+  def accepts(input: String): Boolean = {
+    var currentState = start
+    for (symbol <- input) {
+      val transitionOpt = transitions.find(t => t.from == currentState && t.symbol == symbol)
+      transitionOpt match {
+        case Some(transition) => currentState = transition.to
+        case None => false
+      }
```

Comment:
Our solutions here are very similar - I like that you used symbol <- input for the for -loop. I iterated, and that isn't the most elegant. I also used .filter instead of .find for the transitions, which both seem to result in the same thing, but I wonder if there's a difference.